### DISCUSSION TITLE
Best way to filter the data to be sent to the object storage

### DISCUSSION THREAD

**Discussion by Mojito2110** [thanos-io/thanos#4048] [Questions & Answers]
My deployment includes several Prometheus servers. Each one collects different technical domain metrics. I don't want to load all of the metrics into the object store, but select only a few. 
With the Thanos Sidecar it is not possible to filter, the Prometheus data blocks are loaded in their entirety. 
There are two approaches to filtering if I understand correctly.
first option: set up a Prometheus federation to collect only the metrics from the Prometheus servers that we want to put in the object storage and install the Sidecar on this federation server.
second option : use the remote_write API and a Thanos Receiver and play with the match[] directive in the Prometheus server configuration to select only the metrics I want.
What do you think would be the best solution in terms of scalability, ease of deployment and configuration, scalability etc...

**Comment by yeya24** [comment]
Sorry for the late response. IMO, either way sounds good to me. From the perspective of deployment and configuration, the first option is easier to maintain. As you already have some experience using Sidecar and Prometheus.
The second option is good for future scalability. But it also introduces the receiver clusters. Compared to the first approach, it is not that easy to maintain.
